File: backend/app/Models/Order.php
<?php

namespace App\Models;

use Illuminate\Database\Eloquent\Model;
use Illuminate\Database\Eloquent\Relations\BelongsTo;
use Illuminate\Database\Eloquent\Relations\HasMany;

class Order extends Model
{
    protected $fillable = [
        'order_number',
        'user_id',
        'picker_id',
        'completed_by',
        'courier_id',
        'delivered_by',
        'status',
        'subtotal',
        'delivery_fee',
        'discount',
        'total',
        'payment_method',
        'payment_status',
        'delivery_address',
        'delivery_phone',
        'delivery_name',
        'delivery_time',
        'delivery_type',
        'comment',
        'picking_started_at',
        'picking_completed_at',
        'delivered_at'
    ];

    protected $casts = [
        'subtotal' => 'decimal:2',
        'delivery_fee' => 'decimal:2',
        'discount' => 'decimal:2',
        'total' => 'decimal:2',
        'delivery_time' => 'datetime',
        'picking_started_at' => 'datetime',
        'picking_completed_at' => 'datetime',
        'delivered_at' => 'datetime',
    ];

    public function user(): BelongsTo
    {
        return $this->belongsTo(User::class);
    }

    public function picker(): BelongsTo
    {
        return $this->belongsTo(Picker::class);
    }

    public function completedBy(): BelongsTo
    {
        return $this->belongsTo(Picker::class, 'completed_by');
    }

    public function courier(): BelongsTo
    {
        return $this->belongsTo(Courier::class);
    }

    public function deliveredBy(): BelongsTo
    {
        return $this->belongsTo(Courier::class, 'delivered_by');
    }

    public function address(): BelongsTo
    {
        return $this->belongsTo(Address::class, 'address_id');
    }

    public function items(): HasMany
    {
        return $this->hasMany(OrderItem::class);
    }

    public function getStatusColorAttribute()
    {
        $colors = [
            'processing' => 'warning',
            'accepted' => 'info',
            'preparing' => 'primary',
            'ready' => 'info',
            'delivering' => 'primary',
            'delivered' => 'success',
            'cancelled' => 'danger'
        ];

        return $colors[$this->status] ?? 'secondary';
    }

    public function getStatusNameAttribute()
    {
        $names = [
            'processing' => 'В обработке',
            'accepted' => 'Принят',
            'preparing' => 'Собирается',
            'ready' => 'Собран',
            'delivering' => 'Курьер в пути',
            'delivered' => 'Завершен',
            'cancelled' => 'Отменен'
        ];

        return $names[$this->status] ?? $this->status;
    }

    public function getTotalItemsAttribute()
    {
        return $this->items()->sum('quantity');
    }

    public function getFormattedTotalAttribute()
    {
        return number_format($this->total, 0, '.', ' ') . ' с.';
    }

    public function getFormattedCreatedAtAttribute()
    {
        return $this->created_at->format('d.m.Y H:i');
    }

    protected static function boot()
    {
        parent::boot();
        
        static::creating(function ($order) {
            if (empty($order->order_number)) {
                $order->order_number = 'ORD-' . strtoupper(uniqid());
            }
        });
    }
}

File: lib/models/order.dart
import 'package:flutter/material.dart';

class Order {
  final int id;
  final String orderNumber;
  final int userId;
  final String status;
  final double subtotal;
  final double deliveryFee;
  final double discount;
  final double total;
  final String paymentMethod;
  final String paymentStatus;
  final String deliveryAddress;
  final String deliveryPhone;
  final String? deliveryName;
  final String? deliveryTime;
  final String? deliveryType;
  final String? comment;
  final DateTime createdAt;
  final List<OrderItem> items;

  Order({
    required this.id,
    required this.orderNumber,
    required this.userId,
    required this.status,
    required this.subtotal,
    required this.deliveryFee,
    required this.discount,
    required this.total,
    required this.paymentMethod,
    required this.paymentStatus,
    required this.deliveryAddress,
    required this.deliveryPhone,
    this.deliveryName,
    this.deliveryTime,
    this.deliveryType,
    this.comment,
    required this.createdAt,
    required this.items,
  });

  factory Order.fromJson(Map<String, dynamic> json) {
    return Order(
      id: json['id'],
      orderNumber: json['order_number'],
      userId: json['user_id'],
      status: json['status'],
      subtotal: double.tryParse(json['subtotal'].toString()) ?? 0.0,
      deliveryFee: double.tryParse(json['delivery_fee'].toString()) ?? 0.0,
      discount: double.tryParse(json['discount'].toString()) ?? 0.0,
      total: double.tryParse(json['total'].toString()) ?? 0.0,
      paymentMethod: json['payment_method'],
      paymentStatus: json['payment_status'],
      deliveryAddress: json['delivery_address'],
      deliveryPhone: json['delivery_phone'],
      deliveryName: json['delivery_name'],
      deliveryTime: json['delivery_time'],
      deliveryType: json['delivery_type'],
      comment: json['comment'],
      createdAt: DateTime.parse(json['created_at']),
      items: (json['items'] as List)
          .map((item) => OrderItem.fromJson(item))
          .toList(),
    );
  }

  // Получить локализованное название статуса
  String get statusName {
    switch (status) {
      case 'processing':
        return 'В обработке';
      case 'accepted':
        return 'Принят';
      case 'preparing':
        return 'Собирается';
      case 'ready':
        return 'Собран';
      case 'delivering':
        return 'Курьер в пути';
      case 'delivered':
        return 'Завершен';
      case 'cancelled':
        return 'Отменен';
      default:
        return status;
    }
  }

  // Получить цвет статуса
  Color get statusColor {
    switch (status) {
      case 'processing':
        return Colors.orange;
      case 'accepted':
        return Colors.blue;
      case 'preparing':
        return Colors.purple;
      case 'ready':
        return Colors.teal;
      case 'delivering':
        return Colors.indigo;
      case 'delivered':
        return Colors.green;
      case 'cancelled':
        return Colors.red;
      default:
        return Colors.grey;
    }
  }
}

class OrderItem {
  final int id;
  final int productId;
  final int quantity;
  final double? weight; // вес для весовых товаров (кг)
  final double price;
  final double total;
  final Map<String, dynamic> product;

  OrderItem({
    required this.id,
    required this.productId,
    required this.quantity,
    this.weight,
    required this.price,
    required this.total,
    required this.product,
  });

  factory OrderItem.fromJson(Map<String, dynamic> json) {
    return OrderItem(
      id: json['id'],
      productId: json['product_id'],
      quantity: json['quantity'],
      weight: json['weight'] != null && json['weight'].toString().isNotEmpty
          ? double.tryParse(json['weight'].toString())
          : null,
      price: double.tryParse(json['price'].toString()) ?? 0.0,
      total: double.tryParse(json['total'].toString()) ?? 0.0,
      product: json['product'],
    );
  }
}
